### DISCUSSION TITLE
Examples for https://try.moonbitlang.com/

### DISCUSSION THREAD

**Discussion by StrongTheDev** [moonbitlang/core#1047] [Q&A] [answered]
Hello MoonBitters

I don't know if this is the right place to ask, but i would like to know where the examples repository for [Try MoonBit Language](https://try.moonbitlang.com/) are located. 

I wanted to add some extra code to the examples, specifically the struct example, showing how to define/implement functions for a struct. 

Coming from Rust (just yesterday), it was confusing to see that the online ide from [Try](https://try.moonbitlang.com/) was throwing errors when i was writing this:

```Rust
struct Point {
  x: Int,
  y: Int
}

impl Point {}
```

it would throw them red squiggles 😆 

But as i was reading the MoonBit docs, in the [Enum](https://docs.moonbitlang.com/#enum) section, I came across an eye opener

```Rust
enum Object {
  Point(~x : Double, ~y : Double)
  Circle(~x : Double, ~y : Double, ~radius : Double)
}

...

fn distance_with(self : Object, other : Object) -> Double!NotImplementedError {...}
```
Seeing this made me  realize that we can create struct-specific functions using similar semantics, so I wish to update the example for the `Struct` DataType.

**Comment by gmlewis** [comment]
https://github.com/moonbitlang/moonbit-docs

**Reply by StrongTheDev** [reply]
Thanks, I have checked it out but I have somehow still not been able to find them.

I have checked each folder possible but there is nothing that looks like the examples on the page at [Try MoonBit Lang](try.moonbitlang.com)

**Reply by gmlewis** [reply]
Oh, sorry, I'll let @bobzhang or @peter-jerry-ye answer.

**Comment by Yoorkin** [comment, marked as answer]
Here is the repo: https://github.com/moonbit-community/ide-examples
